AgentInterfaces/VmNetworkConfiguration.cs

```diff
@@ -0,0 +1,29 @@
+﻿namespace Microsoft.Azure.Gaming.AgentInterfaces
+{
+    public class VmNetworkConfiguration
+    {
+        public string PublicIpv4Address { get; set; }
+
+        public string Fqdn { get; set; }
+
+        public Endpoint[] Endpoints { get; set; }
+    }
+
+    public class Endpoint
+    {
+        /// <summary>
+        /// The publicly accessible port exposed on the software load balancer.
+        /// </summary>
+        public int? FrontEndPort { get; set; }
```

Comment:
Why are the ports nullable? Doesn't an endpoint need both a frontEndPort and a backendPort?

Reply:
Good catch. These were copied over from the ComputeNode in CP. Will fix it.